### PR TITLE
Add HeapBlockDevice API docs

### PR DESCRIPTION
**Pull request by kegilbert**
@AnotherButler 
@sg-

### CODE REVIEW

**Comment by kegilbert** [comment]
LGTM, thanks!

File: docs/reference/api/storage/HeapBlockDevice.md [hunk 1]
```diff
@@ -1,2 +1,20 @@
 ## HeapBlockDevice
 
+The HeapBlockDevice class provides a way to simulate block devices for software development or testing. The created blocks are nonvolatile; they do not persist across power cycles.
+
+HeapBlockDevices have some notable configurable parameters:
+
+  - _size_ - Size of the block device in bytes.
```
Comment (sg-) [inline docs/reference/api/storage/HeapBlockDevice.md]: How are these configured? Is this just the constructor? Why are they notable?

Comment (kegilbert) [reply docs/reference/api/storage/HeapBlockDevice.md]: Correct, they're constructor parameters. Wanted to list them out up front, but they are listed in attached class reference links as well. The phrasing is probably a bit off, they're not so much notable parameters as much as a list of parameters used to configure the HeapBlockDevice. I can change the wording or pull that list out.

Comment (AnotherButler) [reply docs/reference/api/storage/HeapBlockDevice.md]: We have a planned section for configuration options for storage APIs (https://jira.arm.com/browse/IOTMORF-1273). However, this planned page should be generated JSON. Would the generated JSON include this content?

**Comment by kegilbert** [comment]
@sg- @AnotherButler 
Follow up changes to emulate the changes in the RawSerial API configuration section.

File: docs/reference/api/storage/HeapBlockDevice.md [hunk 2]
```diff
@@ -1,2 +1,31 @@
 ## HeapBlockDevice
 
+The HeapBlockDevice class provides a way to simulate block devices for software development or testing. The created blocks are nonvolatile; they do not persist across power cycles.
+
+HeapBlockDevices have the following configurable parameters in either one of two constructors:
```
Comment (AnotherButler) [inline docs/reference/api/storage/HeapBlockDevice.md]: Query: What are the two constructors?

Comment (kegilbert) [reply docs/reference/api/storage/HeapBlockDevice.md]: The second block that's talking about creating a block device using a block parameter to set the erase, program, and read sizes is the second constructor. The four parameters above that refer to the first constructor.

Comment (kegilbert) [reply docs/reference/api/storage/HeapBlockDevice.md]: @sg- Added subheaders to break out constructors, are the titles good or should that be rephrased?

**Comment by sg-** [comment]
Seems ok but now I dont understand when to use one or the other or even why both exist...

**Comment by kegilbert** [comment]
@sg- Rephrased it a bit, as far as I can tell the purpose of the first constructor is to just allow more control over erase, program, and read sizes. If they are all bound by the block size then you can just specify that parameter and use the second constructor.

**Comment by sg-** [comment]
Why would there be 2 constructors and not just one with default parameters?

**Comment by kegilbert** [comment]
@sg- Mainly for convenience, having the shortened constructor with just the block_size parameter is easier to use for a lot of block devices that have all three of those parameters restricted to said block size. Using just the verbose constructor with default parameters would be confusing if the only bit of relevant data a user had was block size.

**Comment by kegilbert** [comment]
@sg- Was there anything you wanted changed in this doc?

**Pull request merged**